File: rpg-common/src/main/java/billy/rpg/common/formatter/DialogFormattedResult.java
package billy.rpg.common.formatter;

import java.util.List;
import java.awt.Color;

public class DialogFormattedResult {

    public static class DialogFormattedText {
        public static final DialogFormattedText NEW_LINE = new DialogFormattedText(null, Color.WHITE);

        private String content;
        private Color color;
        private int cnt;

        public DialogFormattedText(String content, Color color) {
            this.content = content;
            this.color = color;
            cnt = content != null ? content.length() : -1;
        }

        public String getContent() {
            return content;
        }

        public Color getColor() {
            return color;
        }

        public int getCnt() {
            return cnt;
        }

        @Override
        public String toString() {
            return content + "@(" + color.getRed() + "," + color.getGreen() + "," + color.getBlue() + ")";
        }
    }

    private int totalLine;
    private List<DialogFormattedText> textList;

    public DialogFormattedResult(int totalLine, List<DialogFormattedText> textList) {
        this.totalLine = totalLine;
        this.textList = textList;
    }

    public int getTotalLine() {
        return totalLine;
    }

    public List<DialogFormattedText> getTextList() {
        return textList;
    }
}

File: rpg-common/src/main/kotlin/billy/rpg/common/formatter/DialogFormattedResult.kt
package billy.rpg.common.formatter

import java.awt.Color

/**
 * @author lei.liu
 * @since 2019-04-16 10:48
 */
data class DialogFormattedResult(val totalLine: Int, val textList: List<DialogFormattedText>)

data class DialogFormattedText(val content: String?, val color: Color) {
    val cnt: Int = content?.length ?: -1

    override fun toString(): String {
        return content + "@(" + color.red + "," + color.green + "," + color.blue + ")"
    }

    companion object {
        val NEW_LINE = DialogFormattedText(null, Color.WHITE)
    }
}
